query/src/main/scala/filodb/query/exec/rangefn/MiscellaneousFunction.scala

```diff
@@ -0,0 +1,95 @@
+package filodb.query.exec.rangefn
+
+import java.util.regex.{Pattern, PatternSyntaxException}
+
+import akka.japi.Option.Some
+import monix.reactive.Observable
+
+import filodb.core.query.{CustomRangeVectorKey, IteratorBackedRangeVector, RangeVector, RangeVectorKey}
+import filodb.memory.format.ZeroCopyUTF8String
+
+trait MiscellaneousFunction {
+  def validator(): Boolean
+
+  def execute(): Observable[RangeVector]
+}
+
+case class LabelReplaceFunction(source: Observable[RangeVector], funcParams: Seq[Any])
+  extends MiscellaneousFunction {
+
+  val labelIdentifier: String= "[a-zA-Z_][a-zA-Z0-9_:\\-\\.]*"
+
+  private def pattern(): Option[Pattern] = {
+    val regex = funcParams(3).asInstanceOf[String]
+    try {
+      val pattern: Pattern = Pattern.compile(regex)
+      Some(pattern)
+    }
+    catch {
+      case ex: PatternSyntaxException => {
+        require(false, "Invalid Regular Expression")
```

Comment:
If PatternSyntaxException has more details of the regex error, it may be useful for user to know. You could simply throw that exception.

Reply:
throwing PatternSyntaxException